### PR TITLE
Check if all linecards are back up when Sup is rebooted

### PR DESCRIPTION
**Pull request by vperumal**
### Description of PR
In the current reboot script, if Supervisor is rebooted, it doesn't verify if all the line cards have come up. This can cause other testcases to fail/error out. Adding a check for supervisor and making sure that all LCs are up.

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
In the current reboot script, if Supervisor is rebooted, it doesn't verify if all the line cards have come up. This can cause other testcases to fail/error out. 

#### How did you do it?
Adding a check for supervisor and making sure that all LCs are up.

#### How did you verify/test it?
Verified on T2 profile

#### Any platform specific information?
No

### CODE REVIEW

**Comment by vperumal** [comment]
FYI @abdosi @gechiang

**Review by abdosi — APPROVED** [review]
LGTM

**Comment by abdosi** [comment]
cc @sanmalho-git . Can you please also review this.

File: tests/platform_tests/test_reboot.py [hunk 1]
```diff
@@ -66,7 +66,19 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list, reboot_type=REB
...
+        logging.error("Failing due to {}".format(e))
+        raise e
+    finally:
+        # If supervisor node is rebooted in chassis, linecards also will reboot.
+        # Check if all linecards are back up.
+        if dut.is_supervisor_node():
+            for host in duthosts:
+                if host != dut:
+                    logging.info("checking if {} critical services are up".format(host.hostname))
+                    wait_critical_processes(host)
```
Comment (sanmalho-git) [inline tests/platform_tests/test_reboot.py:81]: We should also check here that all the interfaces are up. wait_critical_processes I believe only checks if the services have started, but port operationally being up takes longer - specifically as switch create on BRCM J2C+ takes much longer time.

Comment (abdosi) [reply tests/platform_tests/test_reboot.py:81]: @vperumal can you check on this ?

Comment (abdosi) [reply tests/platform_tests/test_reboot.py:81]: @vperumal please see this if this PR is needed

Comment (vperumal) [reply tests/platform_tests/test_reboot.py:97]: @sanmalho-git  and @abdosi  - Added the check

File: tests/platform_tests/test_reboot.py [hunk 2]
```diff
@@ -81,7 +82,26 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list,
...
+    try:
+        check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type)
+    except Exception as e:
+        logging.error("Failing due to {}".format(e))
+        raise e
+    finally:
+        # If supervisor node is rebooted in chassis, linecards also will reboot.
+        # Check if all linecards are back up.
+        if dut.is_supervisor_node():
+            lc_reboot_type = "power off"
```
Comment (wenyiz2021) [inline tests/platform_tests/test_reboot.py:94]: 1. can we also add check for wait_for_startup on the linecards?
2. can we check interfaces and services by reusing `check_interfaces_and_services`?
3. could you please explain why `lc_reboot_type = "power off"`
```
    if dut.is_supervisor_node():
        for lc in duthosts.frontend_nodes:
            wait_for_startup(lc, localhost, delay=10, timeout=300)
            check_interfaces_and_services(lc, interfaces, xcvr_skip_list,
                                          reboot_type=reboot_type)
```

**Review by wenyiz2021 — CHANGES_REQUESTED** [review]

File: tests/platform_tests/test_reboot.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def teardown_module(duthosts, enum_rand_one_per_hwsku_hostname, conn_graph_facts
     check_interfaces_and_services(duthost, interfaces, xcvr_skip_list)
 
 
-def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list,
+def reboot_and_check(localhost, dut, conn_graph_facts, xcvr_skip_list, duthosts,
```
Comment (wenyiz2021) [inline tests/platform_tests/test_reboot.py:59]: changing arguments will cause issue -- please also update the following files that has call to reboot_and_check

platform_tests/test_cont_warm_reboot.py
platform_tests/test_power_off_reboot.py

**Comment by vperumal** [comment]
@wenyiz2021  has created #8856 which is similar to this PR - That PR has been approved, so closing this PR

**Pull request closed without merging**